source/Animate.h

```diff
@@ -0,0 +1,142 @@
+/* Animate.h
+Copyright (c) 2023 by thewierdnut
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef ANIMATE_H_INCLUDED
+#define ANIMATE_H_INCLUDED
+
+
+
+// Smoothly change a variable from one value to another. Used to smooth out
+// scrolling and panning.
+template <typename T>
+class Animate
+{
+public:
+	virtual ~Animate() = default;
+	// Set the next target value of this variable, linearly interpolated using
+	// steps frames.
+	virtual void Set(const T &current, int steps = 5);
+
+	// Reset the pending number of frames to zero. This makes the interpolated
+	// value jump straight to the target value.
+	void ResetAnimation();
```

Comment:
if it is a reset, shouldnt it jump to the original value?

Reply:
The original value is never saved. 

Would a better name work? `ClearAnimation()`? `ResetAnimationState()`?  This function is never used (and it isn't used in the mobile fork where I copied it from either), so It might be better to just delete it.

Reply:
EndAnimation?